### DISCUSSION TITLE
Difference Between opentelemetry-spring-boot-starter and OpenTelemetry Java Agent?

### DISCUSSION THREAD

**Discussion by shahparthiv** [open-telemetry/opentelemetry-java-instrumentation#12904] [Q&A] [answered]
**Description**
I am trying to understand the key differences between the opentelemetry-spring-boot-starter and the OpenTelemetry Java Agent.

**According to my understanding:**

- The main advantage of using the spring-boot-starter is its native image support, making it a good choice for Spring Boot applications that are compiled with tools like GraalVM.

- In terms of instrumentation coverage, both spring-boot-starter and the Java Agent seem to provide similar out-of-the-box support for third-party libraries.

**Questions**

- Is my understanding correct that the main advantage of opentelemetry-spring-boot-starter lies in native image compatibility?

- Do both approaches (starter vs agent) provide the same level of instrumentation coverage for third-party libraries? Or are there differences in supported frameworks and libraries?

- Are there other significant factors (e.g., performance, customizability, etc.) that I should consider when deciding between these two?

**Comment by steverao** [comment, marked as answer]
About the differences between them and when to start to use spring starter. You can refer to https://opentelemetry.io/blog/2024/spring-starter-stable/#when-to-use-the-spring-starter

**Comment by shahparthiv** [comment]
Do we have a similar(Micronaut-starter) thing for micronaut application? To support native image with micronaut application?

**Reply by trask** [reply]
hi @shahparthiv! can you open a new issue for this question so it's not lost here? thanks

**Reply by shahparthiv** [reply]
@trask ,
https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/13052
